lib/GRNOC/TSDS/Setup.pm

```diff
@@ -0,0 +1,887 @@
+package GRNOC::TSDS::Setup;
+
+use Moo;
+
+use GRNOC::TSDS;
+use GRNOC::CLI;
+use GRNOC::Config;
+
+use GRNOC::TSDS::Install;
+
+use Data::Dumper;
+use MongoDB;
+use Sort::Versions;
+use Sys::Hostname;
+### required attributes ###
+
+has config_file => ( is => 'ro',
+    required => 1 );
+
+has unattended => ( is => 'ro',
+    required => 1 );
+
+### internal attributes ###
+
+has cli => ( is => 'ro',
+    default => sub { return GRNOC::CLI->new() } );
+
+has config => ( is => 'rwp' );
+
+has mongo_root => ( is => 'rwp' );
+
+has error => ( is => 'rwp' );
+
+### constructor builder ###
+
+sub BUILD {
+
+    my ( $self ) = @_;
+
+    my $config = GRNOC::Config->new( config_file => $self->config_file,
+        force_array => 0 );
+
+    $self->_set_config( $config );
+
+}
+
+### public methods ###
+
+sub upgrade{
+
+    my ( $self ) = @_;
+
+    $self->cli->clear_screen();
+    $self->_print_banner();
+
+    # -----
+    my $sleep_time = 5;
+    # -----
+
+
+    print "\n Creating mongodb-org-3.6.repo file in /etc/yum.repos.d directory";
+    if(system("(
+            echo [mongodb-org-3.6]
+            echo name=MongoDB Repository
+            echo baseurl=https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/
+            echo gpgcheck=1
+            echo enabled=1
+            echo gpgkey=https://www.mongodb.org/static/pgp/server-3.6.asc
+            )> /etc/yum.repos.d/mongodb-org-3.6.repo")!=0)
+    {
+        print "\n ERROR: Could not write to /etc/yum.repos.d/mongodb-org-3.6.repo";
+        exit;
+    }
+
+    if(system("yum install -y mongodb-org")!=0){
+        print "\n ERROR: Setup failed while installing mongodb-org";
+        exit;
+    }
+
+    print "\n Changing ownership of all files in /var/lib/mongo/ from root to mongod";
+    if(system("chown mongod:mongod /var/lib/mongo/*")!=0){
+        print "\n Error occured while changing ownership of all the files in /var/lib/mongo/ from root to mongod";
+        exit;
+    }
+
+
+    my $is_correct = "n";
+    my $change_ip  = "n";
+    my $hostname;
+    # my $ip;
+    my $ip= "127.0.0.1";
+    while ($is_correct ne "y") {
+
+
+        print "\n Please enter the accurate information below";
+        print "\n Execute 'hostname' command to get the hostname";
```

Comment:
Is there a reason the script doesn't do this itself? Seems like it could suggest the hostname it gets from `hostname`.

Reply:
The reason is that the installation requires a certificate and while creating one (requires input from the user), the field `Common Name` requires the hostname. So if the user already has it copied in the clipboard, it can be pasted directly.